### PR TITLE
Add mpi type

### LINKED ISSUE #50: Enable MPI variants
Enable MPI variants. At the bare minimum, we need to be able to support the current OpenMPI package, as well as a package that would use a native version of MPI.

Ideally, this would fit in well with Anaconda's existing MPI packages.

### PR DESCRIPTION
**Pull request by bethune-bryant**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/master/CONTRIBUTING.md)?
- [x] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/master/doc/)?
- [x] Did you write any [tests](https://github.com/open-ce/open-ce/blob/master/tests/) to validate this change?  

## Description

Fixes #50

### CODE REVIEW

File: doc/README.build_feedstock.md [hunk 1]
```diff
@@ -19,50 +19,56 @@ Command usage for the `build_feedstock.py` command:
...
+                        Comma delimited list of python versions to build for,
+                        such as "3.6" or "3.7". (default: 3.6)
+  --build_types BUILD_TYPES
+                        Comma delimited list of build types, such as "cpu" or
+                        "cuda". (default: cpu,cuda)
+  --mpi_types MPI_TYPES
+                        Comma delimited list of mpi types, such as "openmpi"
+                        or "system". (default: openmpi)
+  --recipe-config-file RECIPE_CONFIG_FILE
+                        Path to the recipe configuration YAML file. The
```
Comment (bnemanich) [inline doc/README.build_feedstock.md:55]: This looks messy now.

Comment (bethune-bryant) [reply doc/README.build_feedstock.md:55]: It looks clean on my end, what looks messy?
image
---
image

Comment (bnemanich) [reply doc/README.build_feedstock.md:55]: The previous section for recipe-config-file used to have new lines in it. This made the example YAML look readable.

Comment (bnemanich) [reply doc/README.build_feedstock.md:55]: Haha, I obviously could have been a little more specific in the previous comment.

**Comment by bethune-bryant** [comment]
This will need to merge first to pass config validation:
open-ce/horovod-feedstock#7

File: open-ce/utils.py [hunk 2]
```diff
@@ -98,6 +117,13 @@ def parse_arg_list(arg_list):
         return arg_list
     return arg_list.split(",") if not arg_list is None else list()
 
+def make_variants(python_versions=DEFAULT_PYTHON_VERS, build_types=DEFAULT_BUILD_TYPES, mpi_types=DEFAULT_MPI_TYPES):
```
Comment (jayfurmanek) [inline open-ce/utils.py:120]: Pulling this here is a good idea.

**Review by jayfurmanek — APPROVED** [review]
This looks good!

**Review by bnemanich — APPROVED** [review]

**Pull request merged**